src/history/hs.c

```diff
@@ -915,7 +917,7 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
         if (clear_hs && (first_non_ts_upd->txnid != list->onpage_upd->txnid ||
                           first_non_ts_upd->start_ts != list->onpage_upd->start_ts)) {
             /* We can only delete history store entries that have timestamps. */
-            WT_ERR(__wt_hs_delete_key_from_ts(session, btree->id, key, 1));
+            WT_ERR(__wt_hs_delete_key_from_ts(session, first_non_ts_upd->txnid, btree->id, key, 1));
```

Comment:
@quchenhao you did the changes that introduced `first_non_ts_upd` and the `oldest_upd` on line 791, could you confirm that those are the correct updates to pull the txnid from?

Reply:
Yes, those look correct to me.

Reply:
Thanks.